pybop/optimisers/scipy_optimisers.py

```diff
@@ -169,11 +174,26 @@ def _run_optimiser(self):
         self.inf_count = 0
 
         # Add callback storing history of parameter values
-        def callback(intermediate_result: OptimizeResult):
-            self.log["x_best"].append(intermediate_result.x)
-            self.log["cost"].append(
-                intermediate_result.fun if self.minimising else -intermediate_result.fun
-            )
+
+        def base_callback(intermediate_result: OptimizeResult):
+            try:
+                self.log["x_best"].append(intermediate_result.x)
+                self.log["cost"].append(
+                    intermediate_result.fun
+                    if self.minimising
+                    else -intermediate_result.fun
+                )
+            except AttributeError:
```

Comment:
Can you add a docstring to this method with information on the try-catch conditions? I.e. when the AttributeError is expected, and what is completed to solve it.